### DISCUSSION TITLE
Are you welcome to features updates?

### DISCUSSION THREAD

**Discussion by leesei** [satnaing/astro-paper#50] [Ideas]
Greate theme, easy to customize.
I've forked and customized it recently, and will continue to customize it for my use case.
Are you welcome to features updates as PR?
Since I've changed quite a lot please check here for a (not-so-formal) change log.
https://leesei.github.io/about/#customizing-astropaper

My fork is [here](https://github.com/leesei/astro-paper), sorry for the lumping of changes :sweat_smile:.
I would like to discuss and see which feature you're interested and I can create a PR for the feature.

**Comment by satnaing** [comment]
Sure.
You can tell me what type of feature(s) you want to add. We can discuss here.

**Comment by leesei** [comment]
My links are included for diffing.
They might not be all the changes, but will be if change is accepted.

## Tier 1

I think these features are useful for most people and not a dramatic change.

- [ ] layout tuning
     https://github.com/leesei/astro-paper/blob/main/src/components/Header.astro
     https://github.com/leesei/astro-paper/blob/main/src/styles/base.css
      increase max width of Header and Footer to `max-w-6xl`
      tune `<Header/>`'s nav (menubar) to `justify-end`
      give more room to site title
- [ ] line break
      https://github.com/remarkjs/remark-breaks
      https://github.com/leesei/astro-paper/blob/main/astro.config.mjs
- [ ] external links in new tab
      https://github.com/rehypejs/rehype-external-links
      https://github.com/leesei/astro-paper/blob/main/astro.config.mjs
- [ ] render `updated` date
      `<Card/>` and `<PostDetails>` shows updated
      https://github.com/leesei/astro-paper/blob/main/src/components/Card.tsx
      https://github.com/leesei/astro-paper/blob/main/src/layouts/PostDetails.astro
      Post lists sort by `updated || created`
      https://github.com/leesei/astro-paper/blob/main/src/utils/getSortedPosts.ts

## Tier 2

These are more personal modifications that fits my needs

- [ ] use `post.slug` (filename) as `/post/{slug}`
     https://github.com/leesei/astro-paper/blob/main/src/layouts/Posts.astro
     https://github.com/leesei/astro-paper/blob/main/src/pages/posts/%5Bslug%5D.astro
     https://github.com/leesei/astro-paper/blob/main/astro.config.mjs
      NOTE: my filenames are already sluggified upon creation
      search also with `post.slug` (filename)
- [ ] support special characters (e.g. `/`, `.`) in tags
      https://github.com/leesei/astro-paper/blob/main/src/utils/tags.ts
      https://github.com/leesei/astro-paper/blob/main/src/pages/tags/index.astro
      https://github.com/leesei/astro-paper/blob/main/src/pages/tags/%5Bslug%5D.astro
      https://github.com/leesei/astro-paper/blob/main/astro.config.mjs
      tags will be slugified by default
      this change effectively introduced `tag_slug`
      add mapping of `tag_slug` <-> `tag` with `getTagsDB()`
- [ ] add `<Tag/>` in `<Card/>`
      convert `<Tag/>` from `.astro` to `.tsx`
- [ ] rewritten `<Breadcrumb/>`   
       https://github.com/leesei/astro-paper/blob/main/src/components/Breadcrumbs.astro
       // see `breadcrumbProps` in these files
       https://github.com/leesei/astro-paper/blob/main/src/layouts/Main.astro
       https://github.com/leesei/astro-paper/blob/main/src/layouts/Posts.astro
       https://github.com/leesei/astro-paper/blob/main/src/pages/search.astro
       https://github.com/leesei/astro-paper/blob/main/src/pages/tags/index.astro
       https://github.com/leesei/astro-paper/blob/main/src/pages/tags/%5Bslug%5D.astro
       use props to determine what to render (what I called context)
       no longer need to parse location
       Breadcrumbs need to specialize in two situations: "tags" and "posts", we can make it accpet other context with default behavior
- [ ] wiki link support
    https://github.com/landakram/remark-wiki-link
    https://github.com/leesei/astro-paper/blob/main/astro.config.mjs
- [ ] use [React Icons](https://react-icons.github.io/react-icons/)
  It makes me easier to introduce other icons, but there are some inconveniences to be overcame for the integration
  https://github.com/leesei/astro-paper/blob/main/src/assets/socialIcons.ts
  https://github.com/leesei/astro-paper/blob/main/src/components/Socials.astro
  https://github.com/leesei/astro-paper/blob/main/src/components/Header.astro
      pick the right icon that can be affected by `{text,fill}-skin-{base,accent}`
      Need to manually specify `/index` upon import, see [here](https://github.com/react-icons/react-icons/issues/509#issuecomment-1470087348)

**Comment by ariestwn** [comment]
@leesei I'm interested to see how you implement the remark-wiki-link. Would you mind sharing the tutorial?

**Reply by leesei** [reply]
https://github.com/leesei/astro-paper/blob/main/astro.config.mjs#L28-L44
This supports `topic#section`, referencing `section` in `topic.md`.

**Comment by eerison** [comment]
I would avoid to add those repositories as dependency, the last commit was 2 years ago, and maybe it could be a block for future upgrades
 - https://github.com/remarkjs/remark-breaks
 - https://github.com/rehypejs/rehype-external-links
